tests/test_pipeline_images.py

```diff
@@ -79,28 +79,29 @@ def test_convert_image(self):
         SIZE = (100, 100)
         # straigh forward case: RGB and JPEG
         COLOUR = (0, 127, 255)
-        im = _create_image('JPEG', 'RGB', SIZE, COLOUR)
-        converted, _ = self.pipeline.convert_image(im)
+        im, buf = _create_image('JPEG', 'RGB', SIZE, COLOUR)
+        converted, converted_buf = self.pipeline.convert_image(im, buf)
         self.assertEqual(converted.mode, 'RGB')
         self.assertEqual(converted.getcolors(), [(10000, COLOUR)])
+        self.assertEqual(converted_buf.read(), buf.read())
```

Comment:
This test is insufficient, because if you comment out the two lines that you added to fix the issue, it passes nonetheless. We need a test that fails before the fix and succeeds afterwards.

You might need to use an actual binary image, as small as possible, instead of creating an image on the fly with Pillow.